File: practice c#/ClassLibrary/Student.cs
using System;

namespace ClassLibrary
{
    public enum Names
    {
        Max,
        Slavik,
        Dima,
        Sasha,
        Vova,
        Vitya,
        Kolya,
        Petya,
        Vasya,
        Misha,
        Yura,
        Roma,
        Oleg,
        Andrey,
        Kirill,
        Denis,
        Nikita,
        Vlad,
        Kamila,
        Dasha,
        Masha,
        Yulya,
        Sveta,
        Nastya,
        Katya,
    }
    public class Student : IComparable<Student>
    {
        public Names Name { get; set; }
        public int Height { get; set; }
        public double Weight { get; set; }

        public Student()
        {
        
        }
        public Student(Names name, int height, double weight)
        {
            Name = name;
            Height = height;
            Weight = weight;
        }

        public override string ToString()
        {
            return $"{Name} , {Height} cm ,{Weight} kg";
        }

        public int CompareTo(Student other)
        {
            if (other == null)
            {
                throw new ArgumentNullException("other", "Student cannot be null");
            }

            // Compare students based on their height
            return Height.CompareTo(other.Height);
        }
    }

}

File: practice java/library/src/Student.java
public class Student implements Comparable<Student> {
    private Names name;
    private int course;
    private char gender;
    public Student() {

    }
    public Student(Names name, int course, char gender) {
        this.name = name;
        this.course = course;
        this.gender = gender;
    }
    public Names getName() {
        return name;
    }
    public void setName(Names name) {
        this.name = name;
    }
    public int getCourse() {
        return course;
    }
    public void setCourse(int course) {
        this.course = course;
    }
    public char getGender() {
        return gender;
    }
    public void setGender(char gender) {
        this.gender = gender;
    }
    @Override
    public String toString() {
        return name + " , " + course + " course  , " + gender;
    }
    @Override
    public int compareTo(Student other) {
        if (other == null) {
            throw new IllegalArgumentException("Student cannot be null");
        }

        return Integer.compare(course, other.getCourse());
    }
}
